drivers/wifi/nrfwifi/CMakeLists.txt

```diff
@@ -0,0 +1,155 @@
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/event.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_api_common.c
+  src/shim.c
+  src/work.c
+  src/timer.c
+  src/fmac_main.c
+  src/fw_load.c
+  src/qspi/src/device.c
+  src/qspi/src/rpu_hw_if.c
+  src/qspi/src/ficr_prog.c
+)
+
+zephyr_library_sources_ifndef(CONFIG_NRF70_RADIO_TEST
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/rx.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_vif.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_util.c
+  src/net_if.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/default/fmac_api.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NET_L2_WIFI_MGMT
+  src/wifi_mgmt_scan.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NRF70_SYSTEM_MODE
+  src/wifi_mgmt.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NRF70_SCAN_ONLY
+  src/wifi_mgmt_scan.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NRF70_RADIO_TEST
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/radio_test/fmac_api.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_util.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NRF70_DATA_TX
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/tx.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_peer.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NRF70_STA_MODE
+  src/wpa_supp_if.c
+  src/wifi_mgmt.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_peer.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_util.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NRF70_AP_MODE
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_ap.c
+)
+
+
+# Without WPA supplicant we only support scan
+zephyr_library_sources_ifdef(CONFIG_NRF70_STA_MODE
+  src/wpa_supp_if.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_peer.c
+  ${OS_AGNOSTIC_BASE}/fw_if/umac_if/src/fmac_util.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NRF70_ON_QSPI
+  src/qspi/src/qspi_if.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NRF70_ON_SPI
+  src/qspi/src/spi_if.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_NRF70_UTIL
+  src/wifi_util.c
+)
+
+zephyr_compile_definitions_ifdef(CONFIG_NRF70_ON_QSPI
+# These are XIP related anomalies and aren't applicable for nRF7002 and cause
+# throughput issues.
+  -DNRF53_ERRATA_43_ENABLE_WORKAROUND=0
+  -DNRF52_ERRATA_215_ENABLE_WORKAROUND=0
+# nRF70 QSPI doesn't use 192MHz clock and most samples use 128MHz, this can cause anomaly 159
+# but as its rare and not seen in most cases, we can disable it.
+# Alternative is 128MHz CPU should be disabled that impacts Wi-Fi performance.
+  -DNRF53_ERRATA_159_ENABLE_WORKAROUND=0
+)
+
+# RPU FW patch binaries based on the selected configuration
+if(CONFIG_NRF70_SYSTEM_MODE)
+  set(NRF70_PATCH ${FW_BINS_BASE}/default/nrf70.bin)
+elseif(CONFIG_NRF70_RADIO_TEST)
+  set(NRF70_PATCH ${FW_BINS_BASE}/radio_test/nrf70.bin)
+elseif(CONFIG_NRF70_SCAN_ONLY)
+  set(NRF70_PATCH ${FW_BINS_BASE}/scan_only/nrf70.bin)
+elseif (CONFIG_NRF70_SYSTEM_WITH_RAW_MODES)
+  set(NRF70_PATCH ${FW_BINS_BASE}/system_with_raw/nrf70.bin)
+else()
+  # Error
+  message(FATAL_ERROR "Unsupported nRF70 patch configuration")
+endif()
+
```

Comment:
Something like
```
if(NOT EXISTS ${NRF70_PATCH})
  message(FATAL_ERROR "
    ------------------------------------------------------------------------
    Missing blobs for NRF70 device driver ... Please install
    using west blobs
    ------------------------------------------------------------------------")
endif()
```

Reply:
Urgh, this checking is causing CI twister failurs, as CI doesn't do `west blobs fetch`, this reminds me of my unsuccessful attempt to add this to the CI, see https://github.com/zephyrproject-rtos/zephyr/pull/59991 

Options:

1. Convert this to a warning, but I guess twister treats warnings as errors?
2. Add `if running_in_ci` but I am not aware of any such check
3. Remove this check and let user's figure out from docs


WDYT?

Reply:
This check should only be invoked if CONFIG_NRF70 is selected :) The CMakeLists.txt should simply not be included in any build which doesn't select it

Reply:
In those failed cases `CONFIG_NRF70` is enabled.

Reply:
Why is CI trying to build this driver?

Reply:
Please remove the sample entry from the wifi sample, that is what is causing CI to try to build the driver, we don't get such luxury when we use blobs :)

Remove this
```
  sample.net.wifi.nordic_nrf70:
    extra_args: SHIELD=nrf7002ek EXTRA_CONF_FILE=overlay-nrf70.conf
    platform_allow:
      - nrf5340dk/nrf5340/cpuapp
      - nucleo_h723zg
    integration_platforms:
      - nrf5340dk/nrf5340/cpuapp
      - nucleo_h723zg
```

Reply:
Sure but what about issue and options in https://github.com/zephyrproject-rtos/zephyr/pull/75714#discussion_r1681639762

Reply:
> 1. Convert this to a warning, but I guess twister treats warnings as errors?

The sample can not be built by CI, so removing this warning will just cause CI to try to build the sample anyway, causing it to fail later. That's what the check is there to avoid.

> 2. Add if running_in_ci but I am not aware of any such check

Removing the entry from sample.yaml does exactly this :) sample.yaml is only there to tell CI to try to build/run specific configurations of the sample.

> 3. Remove this check and let user's figure out from docs

This would be quite hostile to users... We want to help users as much as possible without forcing them to go to Discord or DevZone (where we will then have to explain it again and again...) Checks like this are essential for the user experience.

Reply:
> > 3\. Add if running_in_ci but I am not aware of any such check
> 
> Removing the entry from sample.yaml does exactly this :) sample.yaml is only there to tell CI to try to build/run specific configurations of the sample.

This is also a bad option, as we want to catch any build failures in driver and shields. 



> > 1. Convert this to a warning, but I guess twister treats warnings as errors?
> 
> The sample can not be built by CI, so removing this warning will just cause CI to try to build the sample anyway, causing it to fail later. That's what the check is there to avoid (plus providing the solution of course)

IMHO, this is best option, a warning to end users, but sample will be built to catch other issues. Of course, I need to confirm if twister passes with the warning if not then try another message level,  WDYT?

Reply:
As discussed will remove the commit that adds to twister, will submit a separate PR that enables driver to build without blobs, this can be used for build_only as it cannot be tested without the blobs and helps us catch build issues.

Reply:
I forgot to do the message update, done locally, waiting for other comments/CI and then will push.